src/instructlab/model/chat.py

```diff
@@ -677,19 +671,14 @@ def chat_cli(
     qq,
     greedy_mode,
     max_tokens,
-    tls_insecure,
-    tls_client_cert: Optional[str] = None,
-    tls_client_key: Optional[str] = None,
-    tls_client_passwd: Optional[str] = None,
 ):
     """Starts a CLI-based chat with the server"""
-    cert = get_ssl_cert_config(tls_client_cert, tls_client_key, tls_client_passwd)
```

Comment:
Why not just do:

```python
cart = get_ssl_cert_config(ctx.params["tls_client_cert"], ctx.params["tls_client_key"], ctx.params["tls_client_passwd"])
```

Reply:
The context as all objects in python is passed by reference. So it is more efficient and the code is more clean.

Reply:
Ok glad to know it's a pointer under the hood. What do you mean by non-relevant details? Even though it's passed by reference, I don't like the idea of simply passing `ctx` everywhere will all the details, at least for entrypoint function but not for specific functions like `http_client`.

Reply:
Non-relevant details is what is not relevant for application logic: chat with LLM.
You can treat the `ctx.params` as runtime configuration. Each application module receives access to all configuration and selects what it needs. We can pass only `ctx.params` to `http_client` instead of `ctx` to reduce the scope.